vignettes/data-as-shiny-module.Rmd

```diff
@@ -0,0 +1,127 @@
+---
+title: "Data as shiny Module"
+author: "NEST CoreDev"
+output: 
+  rmarkdown::html_vignette:
+    toc: true
+vignette: >
+  %\VignetteIndexEntry{Data as shiny Module}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+## Introduction
+
+Proper functioning of any `teal` application requires presence of a `teal_data` object.
+Typically, a `teal_data` object created in the global environment will be passed to the `data` argument in `init`.
+This `teal_data` object should contain all elements necessary for successful execution of the application's modules. 
+In some scenarios, however, application developers may opt to postpone some data operations until the application run time.
+This can be done by passing a special _`shiny` module_ to the `data` argument.
+The `teal_data_module` function is used to build such a module from the following components:
+
+- a `ui` function; accepts only one argument, `id`; defines user interface elements for the data module
+- a `server` function: accepts only one argument, `id`; defines server logic for the data module, including data creation; must return a reactive expression containing a `teal_data` object
+
+`teal` will run this module when the application starts and the resulting `teal_data` object that will be used throughout all `teal` (analytic) modules. 
+
+## Creating Data In-App
+
+One case for postponing data operations are data sets that are dynamic, frequently updated.
+Such data cannot be created once and kept in the global environment.
+Using `teal_data_module` allows to create a data set from scratch every time the user starts the application.
+This the user will always have access to the most recent version of the data.
+
+```{r, message = FALSE, warning = FALSE}
+library(teal)
+```
+
+
+```{r}
+data_mod <- teal_data_module(
+  ui = function(id) div(),
+  server = function(id) {
+    moduleServer(id, function(input, output, session) {
+      reactive({
+        data <- within(
+          teal.data::teal_data(),
+          {
+            dataset1 <- iris
+            dataset2 <- mtcars
+          }
+        )
+        teal.data::datanames(data) <- c("dataset1", "dataset2")
```

Comment:
same as in https://github.com/insightsengineering/teal/pull/957/files/422efa6288c33105a6c94ee3681f2f877e5a6344#r1390896679